nilearn/_utils/cache_mixin.py

```diff
@@ -80,17 +80,18 @@ def _check_memory(memory, verbose=0):
     return memory
 
 
+def _nibabel_versions_equal(versions, my_versions):
+    try:
+        return _compare_version(versions, '==', my_versions)
+    except TypeError:
+        return False
+
+
 def _safe_cache(memory, func, **kwargs):
     """A wrapper for mem.cache that flushes the cache if the version
     number of nibabel has changed.
 
     """
-    ''' Workaround for
-     https://github.com/scikit-learn-contrib/imbalanced-learn/issues/482
-    joblib throws a spurious warning with newer scikit-learn.
-    This code uses the recommended method first and the deprecated one
-    if that fails, ensuring th warning is not generated in any case.
-    '''
```

Comment:
As long as you're removing this comment, I would also suggest removing the `AttributeError` block. The original intent of that block was for falling back to `memory.cachedir` when there was no such `memory.location` attribute. However, the code now falls back to `memory.location` if there is no such `memory.location` attribute which doesn't make sense.